index.js

```diff
@@ -29,7 +42,38 @@ async function run() {
     if (!containerDef) {
       throw new Error('Invalid task definition: Could not find container definition with matching name');
     }
-    containerDef.image = imageURI;
+
+    // Check for imageURI
+    if(imageURI) {
+      // Insert the image URI
+      containerDef.image = imageURI;
+    }
+
+    // Check for mergeFile
+    if (mergeFile) {
+      // Parse the merge file
+      const mergeFilePath = path.isAbsolute(mergeFile) ?
+        mergeFile :
+        path.join(process.env.GITHUB_WORKSPACE, mergeFile);
+      if (!fs.existsSync(mergeFilePath)) {
+        throw new Error(`Merge file does not exist: ${mergeFile}`);
+      }
+      const mergeContents = require(mergeFilePath);
+
+      // Merge the merge file
+      if (!Array.isArray(mergeContents.containerDefinitions)) {
+        throw new Error('Invalid merge fragment definition: containerDefinitions section is not present or is not an array');
```

Comment:
I would prefer we not require the presence of `containerDefinitions`, since there are top-level fields which can plausibly differ between environments (task-level memory/CPU, tags, task role, etc.). Is it possible to apply merge functionality across the entire task def vs. just this field?